src/Bicep.Core.Samples/Files/InvalidParameters_LF/main.diagnostics.bicep

```diff
@@ -331,7 +331,7 @@ param paramAccessingResource string = sampleResource
 
 param paramAccessingOutput string = sampleOutput
 //@[06:026) [no-unused-params (Warning)] Parameter "paramAccessingOutput" is declared but never used. (CodeDescription: bicep core(https://aka.ms/bicep/linter/no-unused-params)) |paramAccessingOutput|
-//@[36:048) [BCP058 (Error)] The name "sampleOutput" is an output. Outputs cannot be referenced in expressions. (CodeDescription: none) |sampleOutput|
+//@[36:048) [BCP057 (Error)] The name "sampleOutput" does not exist in the current context. (CodeDescription: none) |sampleOutput|
```

Comment:
Arguably the new error message is lower quality than the original. I don't see any great options for addressing this without adding a bunch of complexity to the binding resolution logic, Ultimately this change means that `metadata` & `output` are simply excluded from the binding resolution logic.

Reply:
`NameBindingVisitor::LookupLocalSymbolByName` could return an `ErrorSymbol` with a diagnostic mentioning that outputs (or metadata, as appropriate) can't be referenced, but I think the new message is fine. We could maybe add a line to the docs calling out that `output` and `metadata` statements don't create symbols if any questions are raised about the new message.

Reply:
The problem with this approach is when you have a valid duplicate:
```bicep
var foo = 'bar'
output foo string = 'bar'

var reference = foo
```

For this scenario to not display an error on `reference`, you need `LookupLocalSymbolByName` to always return the non-error symbol. The fact that the behavior would then differ between cases where there is/isn't a duplicate makes it weird, which then suggests a larger change is needed (for example having `LookupLocalSymbolByName` return a list). This is the point at which I gave up and just decided to go with this approach.